packages/connection-form/src/components/advanced-options-tabs/csfle-tab/csfle-tab.tsx

```diff
@@ -146,6 +157,29 @@ function CSFLETab({
         <Description>
           Specify one or more Key Management Systems to use.
         </Description>
+      </FormFieldContainer>
+      <FormFieldContainer>
+        <Checkbox
+          onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+            handleStoreCredentialsChanged(event.target.checked);
+          }}
+          data-testid="csfle-storeCredentials-input"
+          id="csfle-storeCredentials-input"
+          label={
+            <>
+              <Label htmlFor="csfle-storeCredentials-input">
+                Store KMS provider secrets
+              </Label>
+              <Description>
+                Control whether Compass stores KMS secrets, protected by the OS
```

Comment:
I'm wondering if there's a way, if we find it useful, for us to make this text more explicit that we're either storing or not storing here. Maybe a stretch, but I'm thinking some people might think it's a checkbox for storing in the OS keychain or not in the keychain.
Maybe something like
```
Control whether KMS secrets are stored on disk (protected by the OS keychain) or discarded after disconnecting.
```

Reply:
Sure, done!